wallet/src/libwallet/internal/tx.rs

```diff
@@ -117,25 +86,68 @@ where
 		wallet.keychain(),
 		&mut context.sec_key,
 		&context.sec_nonce,
-		0,
+		participant_id,
 		message,
 	)?;
 
-	Ok((slate, context, sender_lock_fn))
+	Ok((context, sender_lock_fn))
+}
+
+/// Add outputs to the slate, becoming the recipient
+pub fn add_output_to_slate<T: ?Sized, C, K>(
+	wallet: &mut T,
+	slate: &mut Slate,
+	parent_key_id: &Identifier,
+	participant_id: usize,
+	message: Option<String>,
+) -> Result<(Context, OutputLockFn<T, C, K>), Error>
+where
+	T: WalletBackend<C, K>,
+	C: NodeClient,
+	K: Keychain,
+{
+	// create an output using the amount in the slate
+	let (_, mut context, create_fn) =
+		selection::build_recipient_output(wallet, slate, parent_key_id.clone())?;
+
+	// fill public keys
+	let _ = slate.fill_round_1(
+		wallet.keychain(),
+		&mut context.sec_key,
+		&context.sec_nonce,
+		1,
+		message,
+	)?;
+
+	// perform partial sig
+	let _ = slate.fill_round_2(
+		wallet.keychain(),
+		&context.sec_key,
+		&context.sec_nonce,
+		participant_id,
+	)?;
```

Comment:
> And to create a slate as recipient
> create_new_slate()
> add_outputs()

Wouldn't `fill_round_2` return an error in this case, because it checks fee inside, which doesn't make sense for receipt on this stage?

Reply:
Yes, however I haven't gone as far as actually implementing the recipient-initiated transactions yet. Should be easy enough to fix once I get there. (Most likely going to do the wallet split first)